### PR TITLE
correct service principal ID attribute from object_id to id for proper resource referencing

### PR DESCRIPTION
**Pull request by sachincool**
- **chore(terraform): update provider version constraints to allow newer versions**
- **terraform-docs: automated action**
- **chore(versions.tf): update azurerm provider version to 3.107.0 to ensure compatibility with latest features and fixes**
- **fix(cluster_integration.tf): correct service principal ID attribute from object_id to id for proper resource referencing**

### CODE REVIEW

File: cluster_integration.tf [hunk 1]
```diff
@@ -11,7 +11,7 @@ resource "azuread_service_principal" "truefoundry_platform_features_service_prin
 
 resource "azuread_service_principal_password" "truefoundry_platform_features_service_principal_password" {
   count                = var.platform_feature_enabled ? var.feature_cluster_integration_enabled ? 1 : 0 : 0
-  service_principal_id = azuread_service_principal.truefoundry_platform_features_service_principal[0].object_id
+  service_principal_id = azuread_service_principal.truefoundry_platform_features_service_principal[0].id
```
Comment (dunefro) [inline cluster_integration.tf:14]: I don't think this is correct https://registry.terraform.io/providers/hashicorp/azuread/latest/docs/resources/service_principal#object_id 
Just check this one more time

File: versions.tf [hunk 2]
```diff
@@ -4,11 +4,11 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "3.94.0"
+      version = ">= 3.107.0"
     }
     azuread = {
       source  = "hashicorp/azuread"
-      version = "2.47.0"
+      version = ">= 2.47.0"
```
Comment (dunefro) [inline versions.tf:11]: We should move to v3 now

Comment (sachincool) [reply versions.tf:11]: azure erm updated one is at `4.8.0` shall we make a task for this or Should I do it together with my tf changes?

**Review by dunefro — APPROVED** [review]
LGTM

**Pull request merged**